src/monitor-control-service/azext_amcs/custom.py

```diff
@@ -128,14 +129,9 @@ def pre_operations(self):
             else:
                 data = json_data
         for key in data:
-            if key == 'dataSources':
-                args.data_sources = data['dataSources']
-            if key == 'destinations':
-                args.destinations = data['destinations']
-            if key == 'dataFlows':
-                args.data_flows = data['dataFlows']
-            if key == "streamDeclarations":
-                args.stream_declarations = data["streamDeclarations"]
+            arg_key = to_snake_case(key)
+            if hasattr(args, arg_key):
+                setattr(args, arg_key, data[key])
```

Comment:
Can you add tests to verify this solution?

Reply:
Test updated